### PR TITLE
Add support for typeWidening(-preview) reader feature in Delta Lake

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes https://github.com/trinodb/trino/issues/21756

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake
* Add support for reading tables with [type widening](https://docs.delta.io/latest/delta-type-widening.html) feature. ({issue}`21756`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
@findepi Gentle reminder.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java [hunk 1]
```diff
@@ -494,6 +502,21 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
         return new DeltaLakeColumnMetadata(columnMetadata, fieldName, fieldId, physicalName, physicalColumnType);
     }
 
+    private static void verifyTypeChange(JsonNode typeChange)
+    {
+        String fromType = typeChange.get("fromType").asText();
+        String toType = typeChange.get("toType").asText();
+
+        switch ("%s -> %s".formatted(fromType, toType)) {
+            case "byte -> short":
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java:511]: this is a hack we used in tests, but i wouldn't use same pattern in prod code.
let's have direct if on from and to types.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java [hunk 2]
```diff
@@ -494,6 +502,21 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
...
+            case "short -> integer":
+                return;
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java:513]: I can promote byte to  short and then short to integer.
can i promote byte to integer?

**Review by findepi — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java]: We can promote byte to integer. Added another condition and the product test.

**Pull request merged**